Automattic/wp-calypso
# Fix/downgrade offers wrong numbers

#### Proposed Changes

The recent work on cancellation/solutions refactored the Premium -> Personal downgrade path and the Annual -> Monthly downgrade paths. It somehow managed to introduce three problems this PR addresses:

* Premium -> Personal was not visible for annual plans
* Premium -> Personal would show copy from another step that had CURRENT PLAN hardcoded copy and was generally irrelevant
* Premium Monthly -> Personal Monthly doesn't seem to be part of the backend downgrade paths. It seems to work but with incorrect refund amounts and should not be there
* Any plan -> Any plan monthly would show the Personal plan 1y price as the price of the new plan

#### Testing Instructions

* Enable the Store sandbox
* Buy Business 1y
* Go to the cancellation flow and select Price/Too Expensive
* You should see $40 (or local currency equivalent) as the price of the new monthly plan in the proposal
* Cancel the plan
* Buy Premium 1y
* Go to the cancellation flow and select Price/Too Expensive
* You should see a promotion to switch to a more affordable plan with meaningful copy (Personal)

BEFORE:

<img width="1080" alt="Screenshot 2022-12-01 at 14 10 59" src="https://user-images.githubusercontent.com/82778/205062200-ecab6361-5547-4113-b650-55b6b900e1e3.png">

AFTER:
For most annual plans:
<img width="1037" alt="Screenshot 2022-12-01 at 15 09 52" src="https://user-images.githubusercontent.com/82778/205062339-fecac889-9dec-4ba8-9793-d5e52352c6f5.png">

For Premium:
<img width="1045" alt="Screenshot 2022-12-01 at 15 17 37" src="https://user-images.githubusercontent.com/82778/205062615-11ec3241-6a44-43e3-a364-7a4739950956.png">

#### Pre-merge Checklist





Related to #
